acceptance/tests/i18n/check_puppet_run_message.rb

```diff
@@ -8,35 +8,15 @@
   tag 'audit:medium',
       'audit:acceptance'
 
-  def language_name(language)
-    on(agent, 'locale -a') do |locale_result|
-      ["#{language}.utf8", "#{language}.UTF-8", language].each do |lang|
-        return lang if locale_result.stdout =~ /#{lang}/
-      end
-      # no supported language installed skip test on this machine
-      skip_test("test machine is missing #{language} local Skipping")
-    end
-  end
+  require 'puppet/acceptance/i18n_utils'
+  extend Puppet::Acceptance::I18nUtils
 
   language = 'ja_JP'
   agents.each do |agent|
+
     step("ensure #{language} locale is configured") do
-      if agent['platform'] =~ /ubuntu/
-        on(agent, 'locale -a') do |locale_result|
-          if locale_result.stdout !~ /#{language}/
-            on(agent, "locale-gen --lang #{language}")
-            language = language_name(language)
-          end
-        end
-      elsif agent['platform'] =~ /debian/
-        on(agent, 'locale -a') do |locale_result|
-          if locale_result.stdout !~ /#{language}/
-            on(agent, "cp /etc/locale.gen /etc/locale.gen.orig ; sed -e 's/# ja_JP.UTF-8/ja_JP.UTF-8/' /etc/locale.gen.orig > /etc/locale.gen")
-            on(agent, 'locale-gen')
-            language = language_name(language)
-          end
-        end
-      end
+      language = enable_locale_language(agent, language)
+      skip_test("test machine is missing #{language} locale. Skipping") if language.nil?
```

Comment:
In what scenarios do you expect this to return nil? In the utility method I see special cases for Ubuntu and Debian, but in every other cases it just returns its argument (in this case "ja_JP"). Given your report about locales on our test boxes, I would expect this test to be skipped on AIX, for example, but I don't think this will skip it...

Reply:
So on the Ubuntu and Debian machines in the vmpooler, we have to configure the Japanese language locale. If its not listed by 'locale -a', it won't work for the LANGUAGE string and the any test expecting japanese will fail. In at least 1 case I found that I needed to use the longer name ja_JP.utf8 or ja_JP.UTF8, rather than the short name as that is what is listed. Otherwise I'd skip the whole figuring it out step.

I guess I could rewrite it to always try to figure that out where locale(1) is supported, but that will add time to running the tests.

Reply:
The confines at the top appear to take care of the rest.

Reply:
Cool, thanks for clarifying. Should those confines also be added to the third test in this PR then?

Reply:
JK, hidden by the diff. 👍